Add `ECLIPJ2000` as a valid base frame for `ItrsToGcrsRotationModel`

valeriof7 opened the pull request:
Introduce support for `ECLIPJ2000` as a base frame in the `ItrsToGcrsRotationModel` and add corresponding unit tests to ensure consistency between different base frames.

valeriof7 committed:
Add `ECLIP2000` as a valid base frame for `ItrsToGcrsRotationModel`.

Changes to be committed:
- modified:   include/tudat/astro/ephemerides/itrsToGcrsRotationModel.h
- modified:   tests/src/astro/ephemerides/CMakeLists.txt
- modified:   tests/src/astro/ephemerides/unitTestItrsToGcrsRotationModel.cpp

DominicDirkx commented on include/tudat/astro/ephemerides/itrsToGcrsRotationModel.h:
```diff
@@ -75,12 +76,19 @@ class GcrsToItrsRotationModel : public RotationalEphemeris
+                    sofa_interface::getFrameBias( 0.0,
+                                                  anglesCalculator->getPrecessionNutationCalculator( )->getPrecessionNutationTheory( ) );
+        }
+        else if( baseFrame != "J2000" && baseFrame != "ECLIPJ2000" && baseFrame != "GCRS" )
```
Remove the 'GCRS' base frame here. If we ever introduce the GCRS as base frame, we should check whether this model defines the difference with J2000 correctly.

valeriof7 replied:
The "GCRS" is currently the default. Without line 91, the test "test_ItrsToGcrsRotationAgainstSofaCookbook" fails. The error in line 92 will be raised any time the default contractor is used. Any suggestions?

DominicDirkx replied:
Oh, thanks for pointing that out :) Then leave it as it is.

DominicDirkx commented on tests/src/astro/ephemerides/CMakeLists.txt:
```diff
@@ -131,14 +131,17 @@ if(TUDAT_BUILD_WITH_SOFA_INTERFACE)
-            tudat_basic_mathematics
-            tudat_input_output
+            ${Tudat_PROPAGATION_LIBRARIES}
+        #     tudat_simulation_setup
```
You can remove the commented libraries

DominicDirkx commented on tests/src/astro/ephemerides/unitTestItrsToGcrsRotationModel.cpp:
```diff
@@ -178,6 +182,76 @@ BOOST_AUTO_TEST_CASE( test_ItrsToGcrsRotationAgainstSofaCookbook )
+
+    // 5. Comparison in ITRS
+    // Check that the two positions in ITRS are consistent
+    TUDAT_CHECK_MATRIX_CLOSE_FRACTION( marsPositionInItrsFromJ2000, marsPositionInItrsFromEclipj2000, 1.0E-4 );
```
This tolerance is very big (relative error of 1E-4). Check the specific numbers (entry 0,1,2) of the vectors against each other. They should not differ by more than numerical noise

valeriof7 committed:
Resolve PR's comments.

Remove the commented lines from the CMakeLists.txt file and update the unit tests
to check the vector components individually with tighter tolerances.

Changes to be committed:
- modified:   tests/src/astro/ephemerides/CMakeLists.txt
- modified:   tests/src/astro/ephemerides/unitTestItrsToGcrsRotationModel.cpp

DominicDirkx commented:
Merged :)
